src/window/keyboard_manager.rs

```diff
@@ -97,19 +132,27 @@ impl KeyboardManager {
     fn maybe_get_keybinding(&self, key_event: &KeyEvent) -> Option<String> {
         // Determine if this key event represents a key which won't ever
         // present text.
-
         if let Some(key_text) = is_control_key(key_event.logical_key) {
-            Some(self.format_keybinding_string(true, true, key_text))
-        } else {
-            let is_dead_key =
-                key_event.text_with_all_modifiers().is_some() && key_event.text.is_none();
-            let key_text = if (self.alt || is_dead_key) && cfg!(target_os = "macos") {
-                key_event.text_with_all_modifiers()
+            if self.prev_dead_key.is_some() {
+                //recover dead key to normal character
+                let real_char = String::from(self.prev_dead_key.unwrap());
+                Some(real_char + &self.format_keybinding_string(true, true, key_text))
             } else {
+                Some(self.format_keybinding_string(true, true, key_text))
+            }
+        } else {
+            let key_text = if self.prev_dead_key.is_none() {
                 key_event.text
+            } else {
+                key_event.text_with_all_modifiers()
             };
-
-            if let Some(key_text) = key_text {
+            if let Some(ori_key_text) = key_text {
```

Comment:
Super minor, where does the name `ori_key_text` come from?

Reply:
is it original?

Reply:
Looks like "original" to me, i'll fix it in main branch :)